beacon-chain/p2p/peers/status.go

```diff
@@ -62,10 +63,12 @@ var (
 
 // Status is the structure holding the peer status information.
 type Status struct {
-	lock            sync.RWMutex
-	maxBadResponses int
-	status          map[peer.ID]*peerStatus
-	maxLimit        int
+	lock     sync.RWMutex
+	ctx      context.Context
+	cancel   context.CancelFunc
+	status   map[peer.ID]*peerStatus
+	maxLimit int
+	scorer   *PeerScorer
```

Comment:
instead of the scorer being here, it should instead be part of the `peerStatus` object.